flink-runtime/src/test/java/org/apache/flink/runtime/state/StateBackendTestBase.java

```diff
@@ -139,11 +142,25 @@
 	@Rule
 	public final ExpectedException expectedException = ExpectedException.none();
 
+	@Before
+	public void before() {
+		env = buildMockEnv();
+	}
+
+	@After
+	public void after() {
+		IOUtils.closeQuietly(env);
+	}
+
 	// lazily initialized stream storage
 	private CheckpointStreamFactory checkpointStreamFactory;
 
+	private MockEnvironment env;
+
 	protected abstract B getStateBackend() throws Exception;
 
+	protected abstract long getManagedMemorySize();
```

Comment:
Same here with the `getManagedMemorySize`. With the above propose solution we would only need to override this method for RocksDB related classes.